src/SonarScanner.MSBuild.Tasks/Targets/SonarQube.Integration.targets

```diff
@@ -310,6 +323,40 @@
     <MakeDir Directories="$(ProjectSpecificConfDir)" />
   </Target>
 
+  <!-- **************************************************************************** -->
+  <!-- In case of Razor compilation the protobuf files generated during the core compilation step are already moved into a temporary folder.
+       This target moves the protobuf files generated during the razor core compile step into a .razor directory, and alsomoves the protobuf files
+       from the temporary directory back to their original place. This target also takes care about generating the ProjectInfo.xml file for the razor part of the compilation.
+  -->
+  <!-- **************************************************************************** -->
+  <Target Name="CopyProjectInfoFileInPresenceOfRazor" AfterTargets="WriteSonarQubeProjectData" Condition=" $(TemporaryProjectSpecificOutDir) != '' ">
+    <PropertyGroup>
+      <RazorProjectSpecificOutDir>$(ProjectSpecificOutDir).razor</RazorProjectSpecificOutDir>
+    </PropertyGroup>
+    <PropertyGroup>
+      <ProjectInfo>$(ProjectSpecificOutDir)\ProjectInfo.xml</ProjectInfo>
+      <RazorProjectInfo>$(RazorProjectSpecificOutDir)\ProjectInfo.xml</RazorProjectInfo>
+    </PropertyGroup>
+    <!-- Move Razor specific protobuf files to their own directory. -->
+    <MoveDirectory SourceDirectory="$(ProjectSpecificOutDir)"
+               DestinationDirectory="$(RazorProjectSpecificOutDir)" />
+    <!-- Replace the out direcotry in the ProjectInfo.xml for the Razor build to point to the directory with .razor suffix. -->
+    <WriteLinesToFile File="$(RazorProjectInfo)"
+                  Lines="$([System.IO.File]::ReadAllText($(RazorProjectInfo)).Replace('$(ProjectSpecificOutDir)','$(RazorProjectSpecificOutDir)'))"
+                  Overwrite="true"
+                  Encoding="UTF-8"/>
+    <!-- Remove empty ProjectSpecificOutDir to be able to move the content of the temp direcotry. -->
+    <RemoveDir Directories="$(ProjectSpecificOutDir)" />
```

Comment:
This should not be needed from the semantic of this target. We've just moved the directory away in the previous step. There's probably something wrong with the move.
```suggestion
```